core/src/test/java/org/apache/spark/launcher/SparkLauncherSuite.java

```diff
@@ -89,6 +89,11 @@ public void testSparkArgumentHandling() throws Exception {
     launcher.setConf("spark.foo", "foo");
     launcher.addSparkArg(opts.CONF, "spark.foo=bar");
     assertEquals("bar", launcher.builder.conf.get("spark.foo"));
+
+    launcher.setConf(SparkLauncher.PYSPARK_DRIVER_PYTHON, "python3.4");
+    launcher.setConf(SparkLauncher.PYSPARK_PYTHON, "python3.5");
+    assertEquals("python3.4", launcher.builder.conf.get("spark.pyspark.driver.python"));
```

Comment:
It would be better if you checked `PYSPARK_DRIVER_PYTHON.key()`, but I don't know how easy it is to call that from Java. (You could create a tiny scala test class for that in `core/src/test/scala/org/apache/spark/launcher`.)